plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java

```diff
@@ -851,6 +854,106 @@ public void testTableWithNonNullableColumns()
         assertQuery("SELECT * FROM " + tableName, "VALUES(1, null, 100), (2, null, 200)");
     }
 
+    @Test
+    public void testCreateTableWithReaderAndWriterVersion()
+    {
+        String tableName = "test_create_table_with_reader_writer_version_" + randomNameSuffix();
+
+        assertQueryFails(format("CREATE TABLE %s.%s (a_number INT) WITH (reader_version = 0)", SCHEMA, tableName),
+                "reader_version must be an integer between \\[1, 2\\]");
+        assertQueryFails(format("CREATE TABLE %s.%s (a_number INT) WITH (reader_version = 3)", SCHEMA, tableName),
+                "reader_version must be an integer between \\[1, 2\\]");
+        assertQueryFails(format("CREATE TABLE %s.%s (a_number INT) WITH (writer_version = 1)", SCHEMA, tableName),
+                "writer_version must be an integer between \\[2, 4\\]");
+        assertQueryFails(format("CREATE TABLE %s.%s (a_number INT) WITH (writer_version = 5)", SCHEMA, tableName),
+                "writer_version must be an integer between \\[2, 4\\]");
+        assertQueryFails(format("CREATE TABLE %s.%s (a_number INT) WITH (change_data_feed_enabled = true, writer_version = 1)", SCHEMA, tableName),
+                "writer_version must be an integer between \\[2, 4\\]");
+        assertQueryFails(format("CREATE TABLE %s.%s (a_number INT) WITH (change_data_feed_enabled = true, writer_version = 5)", SCHEMA, tableName),
+                "writer_version must be an integer between \\[2, 4\\]");
+
```

Comment:
I'd advocate for separate tests here to have the test method names explaining the functionality tested.